plan/policy.go

```diff
@@ -53,6 +62,17 @@ type CreateOnlyPolicy struct{}
 
 // Apply applies the create-only policy which strips out updates and deletions.
 func (p *CreateOnlyPolicy) Apply(changes *Changes) *Changes {
+	if log.GetLevel() == log.DebugLevel {
```

Comment:
Shell we consider something like that. To avoid code duplication and improve readability, you can extract the logging logic to a separate function.

```
if log.GetLevel() == log.DebugLevel {
	logSkippedEndpoints("deletion", changes.Delete)
	logSkippedEndpoints("update-old", changes.UpdateOld)
	logSkippedEndpoints("update-new", changes.UpdateNew)
}

func logSkippedEndpoints(action string, endpoints []*endpoint.Endpoint) {
	for _, ep := range endpoints {
		log.Debugf(`Skipping %s of endpoint %v due to "create-only" policy`, action, ep)
	}
}
```